demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c

```diff
@@ -0,0 +1,238 @@
+               " Demo." ) );
+
+    /* CK_RV is the return type for a Cryptoki function. Generally the underlying
+     * type is a CK_ULONG, it can also be a CKR_VENDOR_DEFINED type. */
+    CK_RV xResult = CKR_OK;
+
+    /* The CK_FUNCTION_LIST is a structure that contains the Cryptoki version
+     * and a function pointer to each function in the Cryptoki API. If the
+     * function pointer is NULL it is unimplemented. */
+    CK_FUNCTION_LIST_PTR pxFunctionList = NULL;
+
+    /* This Cryptoki library does not implement any initialization arguments. At the time of
+     * writing this demo, the purpose of these optional arguments is to provide
+     * function pointers for mutex operations. */
+    CK_C_INITIALIZE_ARGS xInitArgs = { 0 };
+
+    /* A slot ID is an integer that defines a slot. The Cryptoki definition of
+     * a slot is "A logical reader that potentially contains a token."
+     *
+     * Essentially it is an abstraction for accessing the token. The reason for
+     * this is Some tokens are a physical "card' that needs to be inserted into
+     * a slot for the device to read.
+     *
+     * A concrete example of a slot could be a USB Hardware Security Module (HSM),
+     * which generally appears as a singular slot, and abstracts it's internal "token".
+     *
+     * Some implementations have multiple slots mapped to a single token, or maps
+     * a slot per token. */
+    CK_SLOT_ID * pxSlotId = NULL;
+
+    /* A session is defined to be "The logical connection between an application
+     * and a token."
+     *
+     * The session can either be private or public, and differentiates
+     * your application from the other users of the token. */
+    CK_SESSION_HANDLE hSession = CK_INVALID_HANDLE;
+
+    /* Helper variables. */
+    CK_BYTE xRandomData[ 10 ] = { 0 };
+    uint32_t ulIndex = 0;
+    CK_ULONG xSlotCount = 0;
+
+    /* We use the function list returned by C_GetFunctionList to see what functions
+     * the Cryptoki library supports. We use asserts to ensure that all the
+     * functionality needed in this demo is available. */
+    xResult = C_GetFunctionList( &pxFunctionList );
+    assert( xResult == CKR_OK );
+    assert( pxFunctionList != NULL );
+    assert( pxFunctionList->C_Initialize != NULL );
+    assert( pxFunctionList->C_GetSlotList != NULL );
+    assert( pxFunctionList->C_OpenSession != NULL );
+    assert( pxFunctionList->C_Login != NULL );
+    assert( pxFunctionList->C_GenerateRandom != NULL );
+    assert( pxFunctionList->C_CloseSession != NULL );
+    assert( pxFunctionList->C_Finalize != NULL );
+
+    LogInfo( ( "Cryptoki Major Version: %u Minor Version %u",
+               pxFunctionList->version.major,
+               pxFunctionList->version.minor ) );
+
+    /* C_Initialize will initialize the Cryptoki library and the hardware it
+     * abstracts. */
+    xResult = pxFunctionList->C_Initialize( &xInitArgs );
+    assert( xResult == CKR_OK );
```

Comment:
C-SDK demos do not use asserts but return codes to indicate failure. Is it possible for you to change the style for the demo to be consistent with other demos?

Reply:
Yes I can update to do this